tests/unit/lib/lore_test.sh

```diff
@@ -510,52 +510,57 @@ function test_pre_process_xml_result()
-  expected+='http://lore.kernel.org/amd-gfx/20230809212615.137674-1-davidbtadokoro@usp.br/'
-  assert_equals_helper 'Wrong patchset at index 0' "$LINENO" "$expected" "${list_of_mailinglist_patches[0]}"
-
-  expected='Rodrigo SiqueiraÆrodrigo.siqueira@amd.comÆ1Æ1Ædrm/amdgpu: don'"'"'t allow userspace to create a doorbell BOÆ'
-  expected+='http://lore.kernel.org/amd-gfx/20230809190956.435068-1-rodrigo.siqueira@amd.com/'
-  assert_equals_helper 'Wrong patchset at index 1' "$LINENO" "$expected" "${list_of_mailinglist_patches[1]}"
+  # shellcheck disable=SC2317
+  function is_introduction_patch()
+  {
+    local patch_url="$1"
+    [[ "$patch_url" =~ introduction ]] && return 0
+    return 1
+  }
+
+  # shellcheck disable=SC2317
+  function extract_metadata_from_patch_title()
+  {
+    local patch_title="${1}${SEPARATOR_CHAR}"
+    local patch_url="$2"
+    local patchset_version="X${SEPARATOR_CHAR}"
+    local total_patches="X${SEPARATOR_CHAR}"
+
+    printf '%s%s%s%s' "$patchset_version" "$total_patches" "$patch_title" "$patch_url"
+  }
+
+  # Clear number of patchsets processed and data structure with patchsets
+  reset_current_lore_fetch_session
+
+  # Process first batch of pre processed patches
+  pre_processed_patches_sample=$(< "${SHUNIT_TMPDIR}/samples/pre_processed_patches_sample-1")
+  process_patchsets "$pre_processed_patches_sample"
+  assert_equals_helper 'Wrong number of patchsets processed' "$LINENO" 2 "$PATCHSETS_PROCESSED"
+  expected='Gilberto GilÆgil.gil@mpb.brÆXÆXÆ[PATCH v3] Add Palco to MPBÆhttp://lore.kernel.org/mpb/introduction'
+  assert_equals_helper 'Wrong processed patchset (index 0)' "$LINENO" "$expected" "${list_of_mailinglist_patches[0]}"
+  expected='David BowieÆmajor.tom@rock.ukÆXÆXÆ[RFC PATCH v12] Introduce Ziggy StardustÆhttp://lore.kernel.org/rock/introduction'
+  assert_equals_helper 'Wrong processed patchset (index 1)' "$LINENO" "$expected" "${list_of_mailinglist_patches[1]}"
+
+  # Process first batch of pre processed patches
+  pre_processed_patches_sample=$(< "${SHUNIT_TMPDIR}/samples/pre_processed_patches_sample-2")
+  process_patchsets "$pre_processed_patches_sample"
+  assert_equals_helper 'Wrong number of patchsets processed' "$LINENO" 3 "$PATCHSETS_PROCESSED"
+  expected='Gilberto GilÆgil.gil@mpb.brÆXÆXÆ[PATCH v3] Add Palco to MPBÆhttp://lore.kernel.org/mpb/introduction'
+  assert_equals_helper 'Wrong processed patchset (index 0)' "$LINENO" "$expected" "${list_of_mailinglist_patches[0]}"
+  expected='David BowieÆmajor.tom@rock.ukÆXÆXÆ[RFC PATCH v12] Introduce Ziggy StardustÆhttp://lore.kernel.org/rock/introduction'
+  assert_equals_helper 'Wrong processed patchset (index 1)' "$LINENO" "$expected" "${list_of_mailinglist_patches[1]}"
+  expected='Seu JorgeÆseu.jorge@samba-pop.brÆXÆXÆ[RFC] Release Músicas para Churrasco Vol.1Æhttp://lore.kernel.org/samba-pop/introduction'
+  assert_equals_helper 'Wrong processed patchset (index 2)' "$LINENO" "$expected" "${list_of_mailinglist_patches[2]}"
```

Comment:
Add new lines in the `assert_equals_helper` areas for readability's sake. Maybe consider splitting these checks into two or more test functions?